lib/public-stats.js

```diff
@@ -225,3 +226,37 @@ const updateDailyDealsStats = async (pgClient, committees, findDealClients) => {
     flatStats.map(stat => stat.retrievable)
   ])
 }
+
+/**
+ * @param {pg.Client} pgClient
+ * @param {Iterable<Committee>} committees
+ */
+const updateRetreivalTimes = async (pgClient, committees) => {
+  /** @type {Array<{minerId: string; taskId: string; timeToFirstByteP50: number}>} */
+  const stats = []
+  for (const c of committees) {
+    if (!c.evaluation || !c.evaluation.hasRetrievalMajority || c.evaluation.retrievalResult !== 'OK') continue
+    const { minerId } = c.retrievalTask
+    const taskId = getTaskId(c.retrievalTask)
+    const ttfbMeasurments = []
+    for (const m of c.measurements) {
+      if (m.fraudAssessment !== 'OK') continue
+      ttfbMeasurments.push(m.first_byte_at - m.start_at)
+    }
+
+    const timeToFirstByteP50 = getValueAtPercentile(ttfbMeasurments, 0.5)
+    stats.push({ minerId, taskId, timeToFirstByteP50 })
+  }
+
+  // conflic should never happen, but in case it does we'll ignore the new value
```

Comment:
If conflicts shouldn't happen, I suggest we remove the conflict handling and let it fail. If everything is right, it will never fail. If it fails, it will inform us of a bug to fix.